python/langsmith/_internal/_background_thread.py

```diff
@@ -88,6 +94,39 @@ def _tracing_thread_drain_queue(
     return next_batch
 
 
+def _tracing_thread_drain_compressed_buffer(
+    client: Client,
+    size_limit: int = 100,
+    size_limit_bytes: int = 50 * 1024 * 1024
+) -> Optional[Iterable[bytes]]:
+    with client._buffer_lock:
+        current_size = client.compressed_runs_buffer.tell()
+
+        if not (client._run_count >= size_limit or current_size >= size_limit_bytes):
+            return None
+
+        # Write final boundary and close compression stream
+        client.compressor_writer.write(f'--{client.boundary}--\r\n'.encode())
+        client.compressor_writer.close()
+
+        filled_buffer = client.compressed_runs_buffer
+
+        client.compressed_runs_buffer = io.BytesIO()
+        client.compressor_writer = zstd.ZstdCompressor(level=3).stream_writer(
+            client.compressed_runs_buffer, closefd=False)
+        client._run_count = 0
+
+    filled_buffer.seek(0)
```

Comment:
if we have a buffer we can just pass it directly to requests no? no need for iterator